addons/web/static/src/core/macro.js

```diff
@@ -171,26 +235,26 @@ export class MacroEngine {
         if (this.timeout) {
             browser.clearTimeout(this.timeout);
         }
-        this.timeout = browser.setTimeout(this.advanceMacros.bind(this), this.interval);
+        this.timeout = browser.setTimeout(
+            () => mutex.exec(this.advanceMacros.bind(this)),
+            this.getCheckDelay() || this.defaultCheckDelay
+        );
+    }
+
+    getCheckDelay() {
+        // if a macro has a checkDelay different from 0, use it. Select the maximum.
+        return [...this.macros]
+            .map((m) => m.checkDelay)
+            .filter((delay) => delay > 0)
+            .reduce((m, v) => Math.max(m, v), 0);
```

Comment:
this seems not correct. if i have a slow macro and a fast macro, i want the delay to be the small one, not the large one. If for example knowledge has a macro with a delay of 10ms, you don't want to wait longer because there is a tour running?

Reply:
you're right.